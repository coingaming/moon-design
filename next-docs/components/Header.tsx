import { LogoMoonDesignShort } from '@heathmont/moon-assets';
import { GenericMenu } from '@heathmont/moon-icons-tw';
import Breadcrumbs from './breadcrumbs/Breadcrumbs';

type Props = {
  openSidebar: () => void;
};

const Header: React.FC<Props> = ({ openSidebar }) => (
  <header>
    <div className="relative z-10 py-4 lg:hidden flex flex-row gap-2 items-center justify-between text-bulma">
      <button
        type="button"
        className="focus:outline-none"
        onClick={openSidebar}
      >
        <span className="sr-only">Open sidebar</span>
        <GenericMenu className="text-moon-32" />
      </button>
      <LogoMoonDesignShort fontSize="2rem" />
    </div>
    <div className="relative z-10 hidden lg:block">
      <Breadcrumbs />
    </div>
  </header>
);

export default Header;
