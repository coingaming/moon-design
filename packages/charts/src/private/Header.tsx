import React from 'react';
import styled from 'styled-components';
import { Text } from '@heathmont/moon-components';
import { rem } from '@heathmont/moon-utils';

const Title = styled(Text)(
  ({ theme }) => `
  color: ${theme.color.bulma[100]};
`
);

const FilterWrapper = styled.div``;

const Container = styled.div<{ isActive: boolean }>(
  ({ theme, isActive }) => `
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-bottom: ${rem(theme.space.default)};
  ${
    isActive
      ? `
      ${Title} {
        color: ${theme.color.goten[100]};
      }
      ${FilterWrapper} button * {
        color: ${theme.color.goten[100]};
      }
    `
      : ''
  }
`
);

type Props = {
  icon: React.ReactNode;
  title: string;
  filter?: React.ReactNode;
  isActive?: boolean;
};

export const Header: React.FC<Props> = ({
  icon,
  filter,
  title,
  isActive = false,
}) => {
  return (
    <Container isActive={isActive}>
      {icon}
      <Title size={20}>{title}</Title>
      <FilterWrapper>{filter}</FilterWrapper>
    </Container>
  );
};
