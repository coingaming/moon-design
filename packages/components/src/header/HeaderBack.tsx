import * as React from 'react';
import hideVisually from 'polished/lib/mixins/hideVisually';
import { rem } from '@heathmont/sportsbet-utils';
import { IconArrowLeft } from '@heathmont/sportsbet-assets';
import styled from 'styled-components';

/**
 * Types
 *
 * 1. String or React Component.
 * 2. Allows us to pass in any props associated to the `element`.
 */
/* eslint-disable @typescript-eslint/no-explicit-any */
type HeaderBackProps = {
  element?: any /* [1] */;
  [key: string]: any /* [2] */;
};
/* eslint-enable @typescript-eslint/no-explicit-any */

const HeaderBackLink = styled.a(({ theme: { color, space } }) => ({
  margin: `0 ${rem(space.small)} 0 0`,
  padding: 0,
  alignSelf: 'center',
  verticalAlign: 'middle',
  backgroundColor: 'transparent',
  border: 0,
  appearance: 'none',
  [`> ${IconArrowLeft}`]: {
    color: color.piccolo[100],
    fontSize: rem(space.medium),
  },
}));

/**
 * Component
 */
const HeaderBack: React.FC<HeaderBackProps> = ({
  children,
  element = 'a',
  ...props
}) => (
  <HeaderBackLink as={element} {...props}>
    <IconArrowLeft />
    <span css={hideVisually()}>{children}</span>
  </HeaderBackLink>
);

export default HeaderBack;
