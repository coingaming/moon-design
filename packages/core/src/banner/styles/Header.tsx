import styled from 'styled-components';

type Props = {
  isMultiline?: boolean;
};

const Container = styled.div<Props>(({ isMultiline }) => [
  {
    display: 'grid',
    justifyContent: 'end',
  },
  !isMultiline && {
    gridTemplateColumns: '4fr 1fr',
  },
  isMultiline && {
    gridTemplateRows: '4fr 1fr',
  },
]);

export default Container;
