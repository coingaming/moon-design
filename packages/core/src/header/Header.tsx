import React from 'react';
import { rem } from '@heathmont/moon-utils';
import styled from 'styled-components';

import Container from './styles/Container';
import RightWrapper from './styles/RightWrapper';

export interface HeaderProps {
  backButton?: JSX.Element;
  closeButton?: JSX.Element;
  icons?: JSX.Element[];
  isDivider?: boolean;
}

const Label = styled.div({
  paddingTop: rem(4),
});

const IconWrapper = styled.span({
  marginLeft: rem(8),
});

const BackButtonIcon = styled.div({
  cursor: 'pointer',
  marginRight: rem(8),
  minWidth: rem(32),
});

const ControlsCloseIcon = styled.div({
  cursor: 'pointer',
  marginLeft: rem(8),
});

const Header: React.FC<HeaderProps> = ({
  children,
  backButton,
  closeButton,
  icons,
  isDivider,
}) => (
  <Container isDivider={isDivider}>
    {backButton && <BackButtonIcon>{backButton}</BackButtonIcon>}
    <Label>{children}</Label>
    <RightWrapper>
      {icons &&
        icons.map((item, index) => (
          <IconWrapper key={`hicon-${index}`}>{item}</IconWrapper>
        ))}
      {closeButton && <ControlsCloseIcon>{closeButton}</ControlsCloseIcon>}
    </RightWrapper>
  </Container>
);

export default Header;
