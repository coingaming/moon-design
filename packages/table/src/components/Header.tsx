import {rem, themed} from '@heathmont/moon-utils';
import styled from 'styled-components';

const Header = styled.div<{
  headerBackgroundColor: string,
  selectable?: boolean,
}>(
  {
    position: 'sticky',
    zIndex: 1,
    top: 0,
  },
  ({ selectable, headerBackgroundColor, theme }) => ({
    ...(selectable
      ? {
        paddingLeft: rem(46),
        backgroundColor: themed('color', headerBackgroundColor)(theme)
      } : {})
  })
);

export default Header;
