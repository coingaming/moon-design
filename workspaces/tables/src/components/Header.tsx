import React from 'react';
import type HeaderProps from '../private/types/HeaderProps';

const Header = ({
  headerBackgroundColor,
  selectable,
  children,
}: HeaderProps) => (
  <div
    className="flex flex-col sticky top-0 z-[99]"
    style={
      selectable
        ? { backgroundColor: `rgb(var(--${headerBackgroundColor}))` }
        : {}
    }
  >
    {children}
  </div>
);

export default Header;
