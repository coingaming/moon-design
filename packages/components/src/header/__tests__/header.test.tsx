import React from 'react';
import { create } from 'react-test-renderer';
import { sportsbetDark, ThemeProvider } from '@heathmont/moon-themes';
import 'jest-styled-components';

import Header, { HeaderTitle } from '../Header';
import HeaderBack from '../HeaderBack';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={sportsbetDark}>{component}</ThemeProvider>
);

describe('Header', () => {
  test('renders correctly with title', () => {
    const header = create(
      renderWithTheme(
        <Header>
          <HeaderTitle>Page Title</HeaderTitle>
        </Header>,
      ),
    );

    expect(header).toMatchSnapshot();
  });

  test('renders correctly with title and back button', () => {
    const header = create(
      renderWithTheme(
        <Header>
          <HeaderBack href="../">Back to Cashier</HeaderBack>
          <HeaderTitle>Page Title</HeaderTitle>
        </Header>,
      ),
    );

    expect(header).toMatchSnapshot();
  });
});
