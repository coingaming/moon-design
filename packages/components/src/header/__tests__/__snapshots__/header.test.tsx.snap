// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly with title 1`] = `
.c0 {
  margin-top: 1rem;
  padding-bottom: 0.5rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c1 {
  display: inline-block;
  margin: 0 2rem 0 0;
  font-size: 1.25rem;
  line-height: 1.5625rem;
  font-weight: 600;
}

@media (min-width:31.25em) {
  .c1 {
    font-size: 2rem;
    line-height: 2.4375rem;
  }
}

<header
  className="c0"
>
  <h1
    className="c1"
  >
    Page Title
  </h1>
</header>
`;

exports[`Header renders correctly with title and back button 1`] = `
.c0 {
  margin-top: 1rem;
  padding-bottom: 0.5rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c5 {
  display: inline-block;
  margin: 0 2rem 0 0;
  font-size: 1.25rem;
  line-height: 1.5625rem;
  font-weight: 600;
}

.c3 {
  vertical-align: middle;
}

.c1 {
  margin: 0 0.5rem 0 0;
  padding: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  vertical-align: middle;
  background-color: transparent;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1 > .c2 {
  color: #0CD463;
  font-size: 1.5rem;
}

.c4 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

@media (min-width:31.25em) {
  .c5 {
    font-size: 2rem;
    line-height: 2.4375rem;
  }
}

<header
  className="c0"
>
  <a
    className="c1"
    href="../"
  >
    <svg
      className="c2 c3"
      height="1em"
      viewBox="0 0 20 8"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5.08 8V4.8H20V3.2H5.08V0L0 4z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
    <span
      className="c4"
    >
      Back to Cashier
    </span>
  </a>
  <h1
    className="c5"
  >
    Page Title
  </h1>
</header>
`;
